docs/DeveloperGuide.md

```diff
@@ -4,26 +4,160 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+# 1. Overview
+MeetingJio is a program for **finding potential time slots for team meetings based on everyone’s availability**.
+
+# 2. Design 
+
+## 2.1 Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 2.2 Events
+
+## 2.3 Timetables
+
+## 2.4 Commands
+
+# 3. Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## 3.1 Add Feature
+`add_lesson` allows users to create a new lesson and add it to their own timetables
+
+`add_meeting` allows users to create a new meeting and add it to all the existing timetables
+
+The following sequence diagram shows how the `add_lesson` operation generally works.
+![AddLessonSequenceDiagram](diagrams/AddLessonSequenceDiagram.png)
+
+The following sequence diagram shows how the `add_meeting` operation works in detail.
+![AddMeetingCommandSequence](diagrams/AddMeetingCommandSequence.png)
+
+## 3.2 Listing Events Feature
+The `list` command is a command that the user can input in order to list out the events he has in his timetable.
+
+`list [user]` displays the timetable for the particular user.
+
+`list all` displays the timetable for all users.
+
+In essence, the `list all` command repeatedly calls `list [user]` for all users.
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+The following sequence diagram shows how the command `` is executed.
+![ListCommandSequenceDiagram](diagrams/ListCommandSequenceDiagram.png)
```

Comment:
Diagram formatting is inconsistent. Consider deciding whether to include the bottom row of boxes or not
![image](https://user-images.githubusercontent.com/19286299/160965035-5bd25787-bdf1-4604-8ba6-aaf2f8b4689e.png)
![image](https://user-images.githubusercontent.com/19286299/160965095-d9b24ea1-5e21-4a42-bbb4-a0ce12f5c743.png)